### ISSUE TITLE
Running `flux install anything-else` should return an error

### ISSUE THREAD

**Issue by VinGarcia** [fluxcd/flux2#4403]
### Describe the bug

I recently typed a wrong command when trying to update my local version of flux, I wanted to type:

```
brew install fluxcd/tap/flux
```

but I actually typed:

```
flux install fluxcd/tap/flux
```

And instead of getting an error it worked, which caused the flux-system kustomization to fail, luckly this was on a development cluster, not prod.

Running `flux help install`shows this:

```bash
$ flux help install
The install command deploys Flux in the specified namespace.
If a previous version is installed, then an in-place upgrade will be performed.

Usage:
  flux install [flags]
```

So since it doesn't expect any positional arguments I would have expected it to return an error, which would have saved me some trouble communicating to the guy that manages this (which is not me) about this issue.

### Steps to reproduce

Having any cluster with flux running on it run: `flux install anything`, it works instead of returning an error.

### Expected behavior

It should fail with an error informing that this command expects no positional arguments.

### OS / Distro

macOS 12.5

### Flux version

flux: v2.1.2

### Flux check

$ flux check
► checking prerequisites
✗ Kubernetes version v1.23.16 does not match >=1.25.0-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.36.2
✔ image-automation-controller: deployment ready
► ghcr.io/fluxcd/image-automation-controller:v0.29.0
✔ image-reflector-controller: deployment ready
► ghcr.io/fluxcd/image-reflector-controller:v0.24.0
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v1.1.1
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v1.1.0
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v1.1.2
► checking crds
✔ alerts.notification.toolkit.fluxcd.io/v1beta2
✔ buckets.source.toolkit.fluxcd.io/v1beta2
✔ gitrepositories.source.toolkit.fluxcd.io/v1
✔ helmcharts.source.toolkit.fluxcd.io/v1beta2
✔ helmreleases.helm.toolkit.fluxcd.io/v2beta1
✔ helmrepositories.source.toolkit.fluxcd.io/v1beta2
✔ imagepolicies.image.toolkit.fluxcd.io/v1beta1
✔ imagerepositories.image.toolkit.fluxcd.io/v1beta1
✔ imageupdateautomations.image.toolkit.fluxcd.io/v1beta1
✔ kustomizations.kustomize.toolkit.fluxcd.io/v1
✔ ocirepositories.source.toolkit.fluxcd.io/v1beta2
✔ providers.notification.toolkit.fluxcd.io/v1beta2
✔ receivers.notification.toolkit.fluxcd.io/v1
✗ check failed

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by VinGarcia** [comment]
I just finished writing a PR to address this issue:

https://github.com/fluxcd/flux2/pull/4404

Feel free to change it or propose changes if necessary.

**Comment by hiddeco** [comment]
Fixed by #4404. Thanks you @VinGarcia :bow:
